flow/TLSConfig.actor.cpp

```diff
@@ -741,9 +741,9 @@ std::string getX509Name(const X509_NAME* name) {
 	}
 	X509_NAME_print_ex(out.get(), name, /* indent= */ 0, /* flags */ XN_FLAG_ONELINE);
 	unsigned char* rawName = nullptr;
-	int length = BIO_get_mem_data(out.get(), &rawName);
-	std::string result = (char*)rawName;
-	ASSERT_EQ(result.size(), size_t(length));
+	long length = BIO_get_mem_data(out.get(), &rawName);
+	ASSERT(length > 0);
+	std::string result((const char*)rawName, length);
```

Comment:
This was my suggestion on https://github.com/apple/foundationdb/pull/11409

Reply:
Why don't you use a static_cast? 
```c++
std::string result(static_cast<const char*>(rawName), length);
```
Second are we sure that the returned lenght is correct?
Is better  to lead the ASSERT_EQ at the end..
Are you sure that lenght is signed? Can be unsigned long?